src/components/common/Dialogs/EditDomainDialog/EditDomainDialog.tsx

```diff
@@ -0,0 +1,102 @@
+import React, { useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import { string, object, number, boolean, InferType } from 'yup';
+
+import Dialog, { DialogProps, ActionDialogProps } from '~shared/Dialog';
+import { ActionHookForm as Form } from '~shared/Fields';
+
+import { ActionTypes } from '~redux/index';
+import { WizardDialogType, useEnabledExtensions } from '~hooks';
+import { pipe, withMeta, mapPayload } from '~utils/actions';
+import { DomainColor } from '~gql';
+import { findDomainByNativeId } from '~utils/domains';
+
+import EditDomainDialogForm from './EditDomainDialogForm';
+import { getEditDomainDialogPayload } from './helpers';
+
+interface CustomWizardDialogProps extends ActionDialogProps {
+  filteredDomainId?: number;
+}
+
+type Props = DialogProps &
+  Partial<WizardDialogType<object>> &
+  CustomWizardDialogProps;
+
+const displayName = 'common.EditDomainDialog';
+
+const validationSchema = object()
+  .shape({
+    forceAction: boolean().defined(),
+    domainName: string().max(20).required(),
```

Comment:
Can these have the appropriate messages to avoid the following?

![domainname](https://user-images.githubusercontent.com/64402732/224707511-ce199e5e-3ec1-405c-bf83-915097e56121.png)